tests/e2e/tests/test_prometheus.py

```diff
@@ -0,0 +1,90 @@
+import pytest
+
+from e2e.utils.prometheus.setup_prometheus_server import *
+
+from e2e.utils.constants import DEFAULT_USER_NAMESPACE
+from e2e.utils.utils import (
+    rand_name,
+)
+from e2e.utils.config import metadata
+
+from e2e.conftest import region
+
+from e2e.fixtures.cluster import cluster # Needed for kustomize.
+from e2e.fixtures.kustomize import (
+    kustomize, # Needed for port_forward.
+    clone_upstream, # Needed for kustomize.
+    configure_manifests, # Needed for kustomize.
+)
+from e2e.fixtures.clients import (
+    kfp_client,
+    port_forward, # Needed for kfp_client.
+    host, # Needed for kfp_client.
+    client_namespace, # Needed for kfp_client.
+    session_cookie, # Needed for kfp_client.
+    login, # Needed for session_cookie.
+    password, # Needed for session_cookie.
+)
+
+from kfp_server_api.exceptions import ApiException as KFPApiException
+
+GENERIC_KUSTOMIZE_MANIFEST_PATH = "../../deployments/vanilla"
+
+@pytest.fixture(scope="class")
+def kustomize_path():
+    return GENERIC_KUSTOMIZE_MANIFEST_PATH
+
+class TestPrometheus:
+    @pytest.fixture(scope="class")
+    def setup(self):
+        metadata_file = metadata.to_file()
+        print(metadata.params)  # These needed to be logged
+        print("Created metadata file for TestPrometheus", metadata_file)
+
+    def test_set_up_prometheus(self, metadata, region, kustomize):
+        global workspace_id
+        cluster_name = metadata.get("cluster_name")
+        workspace_id = set_up_prometheus_for_AMP(cluster_name, region)
+        set_up_prometheus_port_forwarding()
```

Comment:
Can this setup be rewritten to be its own fixture rather than a test?
e.g.
```
def prometheus_amp(cluster, region, request, metadata, kustomize):
    metadata_key = "prometheus-amp"
    resource_details = {}

    def on_create():
        workspace_id = set_up_prometheus_for_AMP(cluster, region)
        resource_details["workspace_id"] = workspace_id
    
    def on_delete():
        details = metadata.get(metadata_key)
        wait_for_workspace_deletion(workspace_id=details["workspace_id"])
    
    return configure_resource_fixture(
        metadata=metadata,
        request=request,
        resource_details=resource_details,
        metadata_key=metadata_key,
        on_create=on_create,
        on_delete=on_delete,
    )
```

Few reasons:
    1. Consolidates and isolates set up and tear down of the resource, so that it's lifecycle is easier to understand and the setup/teardown logic will not be interrupted by failures in the setup/teardown of other resources.
    2. Allows the same workspace to be re-used across different test suite runs, which saves time not having to wait for the workspace to be created and also provides a consistent environment to debug in
    3. While pytest runs the tests in the order they show up in the test file, if we want to speed up test runs by parallelizing executions, we will be unable to due to the required ordering present in this file
    4. Aside from the written order of the tests in this file, there is nothing explicit about the order in which the tests will run so we should not write our tests in a way that requires ordering